src/main/java/com/kaleido/fetch/service/ActivityService.java

```diff
@@ -0,0 +1,76 @@
+package com.kaleido.fetch.service;
+
+import com.kaleido.kaptureclient.client.KaptureClient;
+import com.kaleido.kaptureclient.domain.Experiment;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.stereotype.Service;
+import org.springframework.web.bind.annotation.PathVariable;
+
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.Comparator;
+import java.util.List;
+import java.util.concurrent.CompletableFuture;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+@Slf4j
+@Service
+public class ActivityService {
```

Comment:
Should exist a unit test for those Services? I mean, that one and the other you create. :)

Reply:
Agreed this would be a nice addition.